sponnet/pipeline.libsonnet

```diff
@@ -71,6 +75,13 @@
     withUseDefaultArtifact(useDefaultArtifact):: self + { useDefaultArtifact: useDefaultArtifact },
   },
 
+  inputArtifact(id):: {
```

Comment:
would be nice to take a general pass on how sponnet handles artifacts, it's getting a bit convoluted with all the different ways to wrap artifacts

Reply:
This is awesome work and I used this to build all of my pipelines however at this point the template is getting out of sync with spin and does not work with the latest builds.
It would be good to get this merged so the template can be improved to be compliant with spin.
BTW the last version of spin that works with the current template is 9d3a0e2.

Reply:
sorry I was on vacation. I'll take a look at updating this, I also have a few more upstream changes laying around.